server/src/main/java/io/pravega/schemaregistry/storage/StoreExceptions.java

```diff
@@ -97,6 +97,9 @@ public static StoreExceptions create(final Type type, final Throwable cause, fin
             case CONNECTION_ERROR:
                 exception = new StoreConnectionException(errorMessage, cause);
                 break;
+            case AUTH_ERROR:
```

Comment:
this is a call to segment store with delegation token. 
so authorization has already happened at controller. 
so this failure is translated to authenticaiton failure.

Reply:
So, you are saying this really represents that the token failed verification in the segment store and we expect the wire-command to be returning an auth token check failed message: 

https://github.com/pravega/pravega/blob/35385f87a7d8f32ec4de246cdfcd52026f594f97/shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java#L1572

In that case, we should really be throwing a message/exception that indicates the same, rather than Authentication or Authorization. In Pravega main, we had to introduce multiple token related exceptions like `InvalidTokenException`, `TokenExpiredException`, etc. If you want to use a generic one, something like a `TokenException` might make more sense. See the token exception hierarchy here: https://github.com/pravega/pravega/tree/master/shared/authplugin/src/main/java/io/pravega/auth